File: role-builder/add-to-host-queue.rb
require 'beaneater'
require 'json'

beanstalk = Beaneater::Pool.new(['localhost:11300'])
tube = beanstalk.tubes["to-host"]
job = {
  :tar => ARGV[0],
  :appName => ARGV[1],
  :branch => ARGV[2],
  :runCmd => ARGV[3],
  :port => ARGV[4],
  :data => ARGV[5]
}.to_json
tube.put job, :pri => 5
beanstalk.close

File: role-builder/add-to-host-queue.js
console.log('ADDED!');
console.log('$TAR', process.argv[2]);
console.log('$RUNCMD', process.argv[3]);
console.log('$PORT', process.argv[4]);
console.log('$DATAONLY', process.argv[5]);

var amqp = require('amqp');
var amqpConnection = amqp.createConnection({ host: 'dev.rabbitmq.com' });

amqpConnection.publish('to-host', {
	"tar": process.argv[2],
  "runcmd": process.argv[3],
  "port": process.argv[4],
  "dataonly": process.argv[5]
});
